### ISSUE TITLE
Unable to boot 100 clickos instances

### ISSUE THREAD

**Issue by stevepp** [cnplab/clickos#34]
Recently I have been trying to sequentially boot ClickOS. I can not even create 100 clickos instances. 
Starting the 82nd instance, when I try to used clickos-ctl to install a click configuration I got the error "Domain clickos82 does not seem to be a ClickOS domain".

My host has 16GB RAM and 8-core cpu. So in my experiment I fixed 4GB RAM for Xen Dom0 and pin cpus cores 0-2 to Dom0. For each ClickOS vm, I use "xl" to create the vm and then use clickos-ctl to install the configuration.

Does anyone experience the same problem or have any idea why may have caused it?

**Comment by fmanco** [comment]
To indentify a domain as a ClickOS guest, `clickos-ctl` checks for the existence of `/local/domain/<id>/data/clickos`. This entry is created by a ClickOS domain upon boot.

What might be happening here is that the guests are booting slower after a certain point and the entry is not there yet. Have you tried wainting and then run `clickos-ctl` again?

If that doesn't work you need to provide more information. Namely the output of `xl console <guest_82>` and xenstore-ls `/local/domain/<id_guest_82>`.

**Comment by stevepp** [comment]
Sorry to reply so late. 

Yes, I have tried waiting and then run clickos-ctl again. I checked the output of "xl console guest_82" and could see that the main thread has been created. Then checked the output of "xenstore-ls /local/domain/<id_guest_82>". The entry "/local/domain/82/data/clickos" does have been created.

**Comment by fmanco** [comment]
So, to clarify: even though `/local/domain/82/data/clickos` exists at the time you run `clickos-ctl`, it still fails?

**Comment by stevepp** [comment]
Yes, correct. But the weird thing is if I change the domain name from like "ClickOS82" to "p82" in the domain configuration file, then this error message is gone. Do you know why this happens?

**Comment by fmanco** [comment]
Interesting... What is the exact `clickos-ctl` command you use? Do you use a domain name? Maybe I have a bug in the domain name to domain id conversion code.

**Comment by stevepp** [comment]
Here is the command I used to install a click configuration:

"clickos install -s clickos82 clickos82.click"

I guess so. There might be some bug in the code that converts domain name to domain ID.

**Comment by fmanco** [comment]
Can you give the output of `strace -s 512 clickos install -s clickos82 clickos82.click`?
